geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -148,7 +175,7 @@ public long lpush(List<byte[]> elementsToAdd, Region<RedisKey, RedisData> region
    */
   public byte[] lpop(Region<RedisKey, RedisData> region, RedisKey key) {
     byte[] popped = elementRemove(0);
-    RemoveElementsByIndex removed = new RemoveElementsByIndex();
+    RemoveElementsByIndex removed = new RemoveElementsByIndex(incrementAndGetVersion());
```

Comment:
I think this change will conflict with work that's being done for RPOP, since that's also modifying `lpop()` to use versioning properly. This PR might need to wait for RPOP to be merged, then rebase to pick up the changes.